quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/QuarkusKeycloakApplication.java

```diff
@@ -53,16 +68,24 @@ protected void loadConfig() {
 
     @Override
     public Set<Object> getSingletons() {
-        Set<Object> singletons = super.getSingletons().stream()
-                .filter(QuarkusKeycloakApplication::filterSingletons)
-                .collect(Collectors.toSet());
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Set<Class<?>> getClasses() {
+        Set<Class<?>> classes = new HashSet<>(super.getClasses());
+
+        classes.remove(WelcomeResource.class);
+        classes.add(QuarkusWelcomeResource.class);
 
-        singletons.add(new QuarkusWelcomeResource());
+        classes.add(QuarkusObjectMapperResolver.class);
+        classes.add(CloseSessionHandler.class);
+        classes.add(StreamingOutputMessageBodyWriter.class);
```

Comment:
I think this was added to Quarkus as well

Reply:
Yeah, I am thinking of https://github.com/quarkusio/quarkus/pull/30945

Reply:
Seems to work fine without it. Branch updated.

Reply:
@geoand When removing the custom `MessageBodyWriter` we have failures in a test.

For some reason, without the custom `MessageBodyWriter` our exception mapper is not invoked to properly handle exceptions thrown during the deserialization of methods parameters. For instance, we expect our exception mapper setting a proper content type and response payload. See https://github.com/keycloak/keycloak/blob/main/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/error/UncaughtErrorPageTest.java#L253.

Reply:
I'll have to try it tomorrow

Reply:
I did something wrong here.

The difference is actually in how our exception mappers are mapped to exceptions at runtime. I had to add a new `KeycloakMismatchedInputExceptionHandler` to override the default instance set by Resteasy Reactive when mapping exceptions of type `com.fasterxml.jackson.databind.exc.MismatchedInputException`. See https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/ResteasyReactiveJacksonProcessor.java#L112.

We have a generic exception mapper (`org.keycloak.services.error.KeycloakErrorHandler`) to handle `Throwable`. It seems the `DefaultMismatchedInputException` is taking precedence (even though it is not a user-defined mapper) because it is specific for `MismatchedInputException`. Is this behavior expected or Resteasy should run the custom mapper instead?

Reply:
So is there still anything I should look at WRT this?

Reply:
@geoand It should be fine from your side. Thanks for all your help!

Reply:
🙏